### PR TITLE
Make data loading more robust

### PR DESCRIPTION
**Pull request by alanhogan**
- Coerce names to lowercase
- Trim author names

And when something still goes wrong

- show a helpful error message

Note that npm does not allow caps in usernames, so downcasing is fine:
screen shot 2017-09-15 at 6 12 06 am

Here is what some error messages look like:
screen shot 2017-09-15 at 6 43 44 am
screen shot 2017-09-15 at 6 43 17 am

### CODE REVIEW

**Comment by alanhogan** [comment]
*polite bump*

LMK if there is anything I can do to get this merged

**Comment by alanhogan** [comment]
previously failing user entries:

package name: `some-name-that-does-not-exist`

- Used to hang forever
- Now gets a good error message

package name: `YARN` (real package but wrong case)


- Used to hang forever
- Now it just works

user name: ` alanhogan` (valid but with extra whitespace)

- Used to hang forever
- Now it just works

user name: `ALANHOGAN` (valid but wrong case)

- Used to hang forever
- Now it just works

**Comment by alanhogan** [comment]
(updated above comment)

File: src/charts.js [hunk 1]
```diff
@@ -552,13 +552,13 @@ window.submitForm = function submitForm() {
 
         if (packageNames.length >= 1 && packageNames[0].trim() !== '') {
             formData['package'] = $.map(packageNames, function (packageName) {
-                return packageName.trim();
+                return packageName.trim().toLowerCase();
```
Comment (pvorb) [inline src/charts.js]: It's not fine to use `toLowerCase()` here. While it's true that NPM doesn't allow to register new packages using capital letters, this once was possible, e.g. https://www.npmjs.com/package/Set

File: src/charts.js [hunk 2]
```diff
@@ -552,13 +565,13 @@ window.submitForm = function submitForm() {
...
             });
         } else {
             formData['package'] = ['clone'];
         }
     } else if ($nameType.val() == 'author') {
-        var authorName = $('input[name=author]').val();
+        var authorName = $('input[name=author]').val().trim().toLowerCase();
```
Comment (pvorb) [inline src/charts.js]: I don't know whether that's also true for author names. Trimming is fine, though.

File: src/charts.js (hunk 1 above)
Comment (alanhogan) [reply src/charts.js]: 😱  horrifying

File: src/charts.js (hunk 2 above)
Comment (alanhogan) [reply src/charts.js]: Do you think we can try keeping `toLowerCase` here, removing it if anyone files a bug? The fact that npm forces lowercase usernames on sign-up is a pretty big indication that this behavior will be correct most of the time, if not all of the time. If we don’t coerce lowercase here, then additional UI would need added to suggest trying lowercase when a username with mixed or upper case isn't found.

Comment (alanhogan) [reply src/charts.js]: Never mind.. i’ll make the change as requested. 

I emailed NPM Support, who replied:

> Thanks for reaching out.

> You can't assume all usernames are lowercase. Lowercase usernames were disallowed from the namespace approximately two years ago. Prior to that action, it was possible to create mixed-case usernames and some of those accounts may remain in the namespace. We do also know that mixed case usernames are pretty broken.

Sooo yeah

**Comment by alanhogan** [comment]
@pvorb Updated. No longer coerces case.

**Comment by alanhogan** [comment]
@pvorb Sorry for the delay in responding to the lowercase concerns. This should be good to go now.

**Comment by pvorb** [comment]
Thanks for fixing it. I'm currently reworking the deployment process and the overall architecture of the project, so this will still take some time until it goes live.

**Pull request merged**